adapters/utils/runAdapter.ts

```diff
@@ -197,10 +199,28 @@ async function _runAdapter({
         // if (value === undefined || value === null) throw new Error(`Value: ${value} ${key} is undefined or null`)
         if (value instanceof Balances) {
           result[key] = await value.getUSDString()
+          breakdownData[chain] = breakdownData[chain] || {}
+          breakdownData[chain][key] = await value.getUSDJSONs()
           breakdownByToken[chain] = breakdownByToken[chain] || {}
           breakdownByToken[chain][key] = await value.getUSDJSONs()
+          
+          if (value._breakdownBalances) {
+            for (const [label, bal] of Object.entries(value._breakdownBalances)) {
+              const usdString = await (bal as Balances).getUSDString()
+              const breakdownValue = Math.round(+usdString)
+
+              if (!breakdownData[chain]) {
+                breakdownData[chain] = {}
+              }
+              if (!breakdownData[chain][`${key}_breakdown`]) {
+                breakdownData[chain][`${key}_breakdown`] = {}
+              }
+              breakdownData[chain][`${key}_breakdown`][label] = breakdownValue
+            }
+          }
+        } else {
```

Comment:
why is there an else block here? we should always do this to round down the float to integer